src/App.tsx

```diff
@@ -149,6 +147,16 @@ const App = (props: AppProps) => {
       .catch((err) => console.error(err));
   }
 
+  function updateMaxDisplayedCycles(newValue: number) {
```

Comment:
First, please don't use `function` notation here and use arrow function instead
Second, you can add `async` for this function and avoid `then/catch`
Also you can wrap this function with `useCallback` hook